apps/elmo_queues/consumers.py

```diff
@@ -0,0 +1,86 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+'''Consume data from pulse and insert it into the database.
+'''
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import logging
+
+from django.conf import settings
+from django.db import connection
+
+from kombu import Queue, Exchange
+from kombu.mixins import ConsumerMixin
+
+from life.models import Forest, Repository
+from pushes import utils
+
+
+HGMO = Queue(
+    f"queue/{settings.PULSE_USER}/hgmo",
+    routing_key="#", expires=settings.PULSE_TTL,
+    exchange=Exchange(
+        "exchange/hgpushes/v2",
+        durable=True, auto_delete=False, type="topic", no_declare=True,
```

Comment:
why `no_declare`?

Reply:
Good question. Honest answer is "copy and paste" ;-)

Reading https://docs.celeryproject.org/projects/kombu/en/stable/reference/kombu.html#kombu.Exchange.declare, it seems I want `passive=True` here?

Reply:
The typical approach is for everything that uses a queue to declare it (not passively).  If it already exists, no problem.  If not, it is created.  It avoids any need for manual creation of the queue.

Reply:
If the queue does not exist and is created, is there a way for us to know?

The case I'm concerned about is something going wrong causing this service not consuming messages, pulseguardian deleting the queue because it has too many messages, the service restarting, and the queue being recreated without us realizing we missed messages.

I guess that exact chain of events is quite unlikely to happen because the number of messages published to this exchange is low enough that we could go longer than 24 hours without processing any without triggering pulseguardian. Even over the weekend I'd get alerted if Elmo didn't do any work for 24 hours.

Reply:
From Axel in slack:

> also found the numbers of messages: 2k for a  warning email, which is 1.5-2 days in practice, I guess. At 8k, the queue is taken down

so I'm no longer concerned about autocreation of queue.

Reply:
The queue is `no_declare=False`, and comes up on start.

Probably talked pass each other on the manual creation: We need the queue to be up before we take down the poller, and we'll need to kick that off somewhere outside of the deployment.

The passive part is only on the Exchange, and I think that's OK?

Reply:
OK, sounds good. You understand the situation more fully than I do, and your approach is solid.

Reply:
Also, I read https://bugzilla.mozilla.org/show_bug.cgi?id=1653606#c1 that telegraf monitoring we can white-list the queue to be not deleted by pulseguardian?

Reply:
That's my understanding :)